README.md

```diff
@@ -1,4 +1,1060 @@
+  };
+};
+```
+
+It is of course possible to add other inputs (not already managed by IOGX) in the normal way. 
+
+For example, to add `nix2container` and `cardano-world`:
+
+```nix
+inputs = {
+  iogx.url = "github:inputs-output-hk/iogx";
+  n2c.url = "github:nlewo/nix2container";
+  cardano-world.url = "github:input-output-hk/cardano-world";
+};
+```
+Note that IOGX will merge (union) its implicit inputs and the new inputs (`n2c`, `cardano-world`) into a single attrset, which will simply be called `inputs` in the API, and will be available to you as a function parameter in most files in the [`nix` folder](./template/nix).
+
+In conclusion, the `inputs` parameter will contain both IOGX inputs and yours.
+
+### 3.1.3. `outputs`
+
+This line is boilerplate and should not be changed. 
+
+IOGX hosts its main `mkFlake` function in the `lib` top-level attribute. 
+
+There are other functions in `lib`, but they are not needed to use IOGX and will be documented later.
+
+Note that the call to `mkFlake` must take a second argument (`./.`), but this restriction will be lifted soon.
+
+As stated in the [Introduction](#1-introduction), your final [flake outputs](#310-flake-outputs-format) are generated based on the contents of your [`nix`](./template/nix) folder.
+
+### 3.1.4. `nixConfig`
+
+Unless you know what you are doing, you should not change `nixConfig`.
+
+You could always add new `extra-substituters` and `extra-trusted-public-keys`, but do not delete the existing ones, or you won't have access to IOG caches. 
+
+For the caches to work properly, it is sufficient that the following two lines are included in your `/etc/nix/nix.conf`:
+```txt
+trusted-users = USER
+experimental-features = nix-command flakes
+```
+Replace `USER` with the result of running `whoami`. 
+
+You may need to reload the nix daemon on Darwin for changes to `/etc/nix/nix.conf` to take effect:
+```bash
+sudo launchctl stop org.nixos.nix-daemon
+sudo launchctl start org.nixos.nix-daemon
+```
+Leave `allow-import-from-derivation` set to `true` for `haskell.nix` for work correctly.
+
+## 3.2. `nix/iogx-config.nix`
+
+```nix
+# Example of a valid nix/iogx-config.nix
+{ 
+  repoRoot = ../.; 
+  systems = [ "x86_64-linux" "x86_64-darwin" ]; 
+  haskellCompilers = [ "ghc8107" ]; 
+  defaultHaskellCompiler = "ghc8107"; 
+  shouldCrossCompile = true; 
+} 
+```
+
+This file must exist and it contains the main configuration values for your project.
+
+### 3.2.1. `repoRoot` 
+
+A Nix path to an existing folder containing your `cabal.project` file.
+
+In future versions this field will be removed and it will default to the repository top-level directory, but until then it must be explicitly set (usually to `../.`).
+
+### 3.2.2. `systems`
+
+The non-empty list of systems against which your project can be built. 
+
+These are standard Nix values found in `nixpkgs.stdenv.system`.
+
+This field affects your final [flake outputs](#310-flake-outputs-format).
+
+This field is required.
```

Comment:
IMO defaulting to Linux, and then adding in the two Darwin systems once Hydra has hardware, is fine.